### PR TITLE
Kafka ssl support

### PR DESCRIPTION
**Pull request by Mixton**
Add SSL configuration options for Kafka clients as described in Shopify/sarama documentation.

### CODE REVIEW

File: input/kafkamdm/kafkamdm.go [hunk 1]
```diff
@@ -77,6 +82,10 @@ func ConfigSetup() {
 	inKafkaMdm.DurationVar(&consumerMaxWaitTime, "consumer-max-wait-time", time.Second, "The maximum amount of time the broker will wait for Consumer.Fetch.Min bytes to become available before it returns fewer than that anyway")
 	inKafkaMdm.DurationVar(&consumerMaxProcessingTime, "consumer-max-processing-time", time.Second, "The maximum amount of time the consumer expects a message takes to process")
 	inKafkaMdm.IntVar(&netMaxOpenRequests, "net-max-open-requests", 100, "How many outstanding requests a connection is allowed to have before sending on it blocks")
+	inKafkaMdm.BoolVar(&tlsEnabled, "tls-enabled", false, "Whether to enable TLS")
+	inKafkaMdm.BoolVar(&tlsSkipVerify, "tls-skip-verify", false, "Whether skip TLS server cert verification")
+	inKafkaMdm.StringVar(&tlsClientCert, "tls-client-cert", "", "Client cert for client authentication (use with -tls-enabled and -tls-client-key)")
+	inKafkaMdm.StringVar(&tlsClientKey, "tls-client-key", "", "Client key for client authentication (use with tls-enabled and -tls-client-cert)")
```
Comment (replay) [inline input/kafkamdm/kafkamdm.go]:
```suggestion
	inKafkaMdm.StringVar(&tlsClientKey, "tls-client-key", "", "Client key for client authentication (use with -tls-enabled and -tls-client-cert)")
```

File: mdata/notifierKafka/cfg.go [hunk 2]
```diff
@@ -47,6 +52,10 @@ func init() {
 	FlagSet.StringVar(&partitionStr, "partitions", "*", "kafka partitions to consume. use '*' or a comma separated list of id's. This should match the partitions used for kafka-mdm-in")
 	FlagSet.StringVar(&offsetStr, "offset", "newest", "Set the offset to start consuming from. Can be oldest, newest or a time duration")
 	FlagSet.StringVar(&backlogProcessTimeoutStr, "backlog-process-timeout", "60s", "Maximum time backlog processing can block during metrictank startup. Setting to a low value may result in data loss")
+	FlagSet.BoolVar(&tlsEnabled, "tls-enabled", false, "Whether to enable TLS")
+	FlagSet.BoolVar(&tlsSkipVerify, "tls-skip-verify", false, "Whether skip TLS server cert verification")
+	FlagSet.StringVar(&tlsClientCert, "tls-client-cert", "", "Client cert for client authentication (use with -tls-enabled and -tls-client-key)")
+	FlagSet.StringVar(&tlsClientKey, "tls-client-key", "", "Client key for client authentication (use with tls-enabled and -tls-client-cert)")
```
Comment (replay) [inline mdata/notifierKafka/cfg.go]:
```suggestion
	FlagSet.StringVar(&tlsClientKey, "tls-client-key", "", "Client key for client authentication (use with -tls-enabled and -tls-client-cert)")
```

File: input/kafkamdm/kafkamdm.go [hunk 3]
```diff
@@ -77,6 +82,10 @@ func ConfigSetup() {
 	inKafkaMdm.DurationVar(&consumerMaxWaitTime, "consumer-max-wait-time", time.Second, "The maximum amount of time the broker will wait for Consumer.Fetch.Min bytes to become available before it returns fewer than that anyway")
 	inKafkaMdm.DurationVar(&consumerMaxProcessingTime, "consumer-max-processing-time", time.Second, "The maximum amount of time the consumer expects a message takes to process")
 	inKafkaMdm.IntVar(&netMaxOpenRequests, "net-max-open-requests", 100, "How many outstanding requests a connection is allowed to have before sending on it blocks")
+	inKafkaMdm.BoolVar(&tlsEnabled, "tls-enabled", false, "Whether to enable TLS")
+	inKafkaMdm.BoolVar(&tlsSkipVerify, "tls-skip-verify", false, "Whether skip TLS server cert verification")
```
Comment (replay) [inline input/kafkamdm/kafkamdm.go]:
```suggestion
	inKafkaMdm.BoolVar(&tlsSkipVerify, "tls-skip-verify", false, "Whether to skip TLS server cert verification")
```

File: mdata/notifierKafka/cfg.go [hunk 4]
```diff
@@ -47,6 +52,10 @@ func init() {
 	FlagSet.StringVar(&partitionStr, "partitions", "*", "kafka partitions to consume. use '*' or a comma separated list of id's. This should match the partitions used for kafka-mdm-in")
 	FlagSet.StringVar(&offsetStr, "offset", "newest", "Set the offset to start consuming from. Can be oldest, newest or a time duration")
 	FlagSet.StringVar(&backlogProcessTimeoutStr, "backlog-process-timeout", "60s", "Maximum time backlog processing can block during metrictank startup. Setting to a low value may result in data loss")
+	FlagSet.BoolVar(&tlsEnabled, "tls-enabled", false, "Whether to enable TLS")
+	FlagSet.BoolVar(&tlsSkipVerify, "tls-skip-verify", false, "Whether skip TLS server cert verification")
```
Comment (replay) [inline mdata/notifierKafka/cfg.go]:
```suggestion
	FlagSet.BoolVar(&tlsSkipVerify, "tls-skip-verify", false, "Whether to skip TLS server cert verification")
```

**Comment by Mixton** [comment]
Hi Mauro,
Thanks for your time. Your changes was added and the conf/ini files updated. Ci is in progress. Let me know if you need something else.

**Comment by replay** [comment]
thx @Mixton

**Pull request merged**